include/hal/library/responder/key_pair_info.h

```diff
@@ -0,0 +1,53 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2024 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
+ **/
+
+#ifndef RESPONDER_KEY_PAIR_INFO_H
+#define RESPONDER_KEY_PAIR_INFO_H
+
+#include "hal/base.h"
+#include "internal/libspdm_lib_config.h"
+#include "library/spdm_return_status.h"
+#include "industry_standard/spdm.h"
+
+#if LIBSPDM_ENABLE_CAPABILITY_GET_KEY_PAIR_INFO_CAP
+
+/**
+ * read the key pair info of the key_pair_id.
+ *
+ * @param  spdm_context               A pointer to the SPDM context.
+ * @param  key_pair_id                Indicate which key pair ID's information to retrieve.
+ * @param  base_asym_algo             Indicates the signing algorithm.
+ *
+ * @param  total_key_pairs            Indicate the total number of key pairs on the responder.
+ * @param  capabilities               Indicate the capabilities of the requested key pairs.
+ * @param  key_usage_capabilities     Indicate the key usages the responder allows.
+ * @param  current_key_usage          Indicate the currently configured key usage for the requested key pairs ID.
+ * @param  asym_algo_capabilities     Indicate the asymmetric algorithms the Responder supports for this key pair ID.
+ * @param  current_asym_algo          Indicate the currently configured asymmetric algorithm for this key pair ID..
+ * @param  public_key_info_len        On input, indicate the size in bytes of the destination buffer to store.
```

Comment:
Swap position of `public_key_info_len` and `assoc_cert_slot_mask`. `public_key_info_len` and `public_key_info` should be adjacent to one another.

Reply:
Thanks. The previous design is followed the SPDM spec. Now I have changed the design to follow your feedback.

![image](https://github.com/user-attachments/assets/6863d364-16f2-4104-9f27-48591a899802)